src/chatd.cpp

```diff
@@ -1735,20 +1735,25 @@ void Chat::requestRichLink(Message &message)
                 for (std::string::size_type i = 0; i < originalMessage.size(); i++)
                 {
                     char character = originalMessage[i];
-                    if (character != '\n' && character != '\r')
-                    {
-                        textMessage.push_back(originalMessage[i]);
-                    }
-                    else if (character == '\n')
+                    if (character == '\n')
                     {
                         textMessage.push_back('\\');
                         textMessage.push_back('n');
                     }
-                    else
+                    else if (character == '\r')
                     {
                         textMessage.push_back('\\');
                         textMessage.push_back('r');
                     }
+                    else if (character == '"')
```

Comment:
We should also escape `‘` (single-quote), `\` (backslash) and all the control characters (ASCII codes from 0 to 31).